posts/2022-10-27-22.0.0.12-beta.adoc

```diff
@@ -0,0 +1,351 @@
+
+Previously, imported packages and classes were only available to the scripting environment (i.e. scriptlet tags). However, with version 3.1, they are also available in the Expression Language environment within the page. Default imports (such as `jakarta.servlet`, `jakarta.servlet.jsp`, `jakarta.servlet.jsp.http`) are consistent between both environments. Related changes included internal refactoring of the `ScopedAttributeELResolver` to create `ImportELResolver` and the addition of the `NotFoundELResolver`, which always resolves the requested value and returns `null`.
+
+Other significant changes include the deprecation of the `isThreadSafe` directive, the `jsp:plugin` (including the related `jsp:params` and `jsp:fallback`) actions, and any methods which overrode `ELResolver.getFeatureDescriptors()`.
+
+As for Jakarta Tags 3.0, the largest change includes the Tag URI renaming when referencing tag libraries. For example, `jakarta.tags.core` now replaces the `http://xmlns.jcp.org/jsp/jstl/core` URI when importing the core tag library.  However, the implementation kept older URIs for backwards compatibility.  The remaining changes related to general clean up of the documentation.
+      
+
+To enable the new Jakarta Server Pages 3.1 and Jakarta Standard Tag Library 3.0 features, add the `pages-3.1` feature to your server.xml:
+[source,xml]
+---- 
+<featureManager>
+   <feature>pages-3.1</feature>
+</featureManager>
+---- 
+ 
+For more information about Jakarta Server Pages and Jakarta Standard Tag Library, refer to the following links:  
+
+* link:{url-prefix}/docs/latest/reference/feature/pages.html[Feature Documentation: Jakarta Server Pages]
+* link:https://jakarta.ee/specifications/pages/3.1/[Specification: Jakarta Server Pages 3.1]
+* link:https://jakarta.ee/specifications/tags/3.0/[Specification: Jakarta Standard Tag Library 3.0]
+* link:https://jakarta.ee/specifications/pages/3.1/apidocs/jakarta.servlet.jsp/module-summary.html[Javadoc: Jakarta Server Pages 3.1]
+* link:https://jakarta.ee/specifications/tags/3.0/apidocs/jakarta.servlet.jsp.jstl/module-summary.html[Javadoc: Jakarta Standard Tag Library 3.0]
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 
+
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/22416
+// Contact/Reviewer: volosied,pnicolucci,ReeceNana
+// // // // // // // // 
+[#websocket]
+== Register endpoints outside of the deployment phase (Jakarta WebSocket 2.1)
+
+Jakarta WebSocket allows communication via the WebSocket protocol, allowing for full-duplex communication between endpoints.  This means that, unlike the HTTP protocol, a single, persisted connection is made across the endpoints, that allows the server to continuously push information to the client. 

```

Comment:
Don't like this wording after another read through. Can we change it to: 

This means that a single, persisted connection is made across the endpoints, allowing both the server and client to continuously and simultaneously push information to each other.

Reply:
I was trying to focus on the differentiation from HTTP, but this makes sense